src/Components/Matrix/DrawMatrix.jsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import "./Matrix.css";
+import { useSelector } from "react-redux";
+import { incrementCellFC, setAverage } from "../../slices/matrixSlice";
+
+const DrawMatrix = ({ matrix, findAverage, dispatch}) => {
```

Comment:
do not use dispatch as prop.
Just call this hook again into this component.
`const dispatch = useDispatch();`